src/main/java/com/mercateo/rest/jersey/utils/listing/SearchQueryParameterBean.java

```diff
@@ -15,58 +15,78 @@
  */
 package com.mercateo.rest.jersey.utils.listing;
 
+import com.mercateo.common.rest.schemagen.IgnoreInRestSchema;
+
+import javax.validation.constraints.Min;
 import javax.ws.rs.DefaultValue;
 import javax.ws.rs.QueryParam;
 
-import com.mercateo.common.rest.schemagen.IgnoreInRestSchema;
-
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.Setter;
 
 @NoArgsConstructor
 public class SearchQueryParameterBean {
-	@Getter
-	@Setter
-	@IgnoreInRestSchema
-	private int defaultLimit = 20;
-
-	@IgnoreInRestSchema
-	private int defaultOffset = 0;
-
-	@QueryParam("offset")
-	@DefaultValue("0")
-	private Integer offset;
-
-	@QueryParam("limit")
-	private Integer limit;
-
-	public SearchQueryParameterBean(int offset, int limit) {
-		super();
-		this.offset = Integer.valueOf(offset);
-		this.limit = Integer.valueOf(limit);
-	}
-
-	public int getOffset() {
-		if (offset == null) {
-			return defaultOffset;
-		}
-		return offset.intValue();
-	}
-
-	public void setOffset(int offset) {
-		this.offset = Integer.valueOf(offset);
-	}
-
-	public int getLimit() {
-		if (limit == null) {
-			return defaultLimit;
-		}
-		return limit.intValue();
-	}
-
-	public void setLimit(int limit) {
-		this.limit = Integer.valueOf(limit);
-	}
+
+    @Getter
+    @Setter
+    @IgnoreInRestSchema
+    private int defaultLimit = 20;
+
+    @IgnoreInRestSchema
+    private int defaultOffset = 0;
+
+    @QueryParam("offset")
+    @DefaultValue("0")
+    @Min(0)
+    private Integer offset;
+
+    @QueryParam("limit")
+    @Min(0)
+    private Integer limit;
+
+    public SearchQueryParameterBean(int offset, int limit) {
+        super();
+        checkOffset(offset);
+        this.offset = Integer.valueOf(offset);
+        checkLimit(limit);
+        this.limit = Integer.valueOf(limit);
+    }
+
+    public int getOffset() {
+        if (offset == null) {
+            return defaultOffset;
+        }
+        return offset.intValue();
+    }
+
+    public void setOffset(int offset) {
+        checkOffset(offset);
+        this.offset = Integer.valueOf(offset);
+    }
+
+    public int getLimit() {
+        if (limit == null) {
+            return defaultLimit;
+        }
+        return limit.intValue();
+    }
+
+    public void setLimit(int limit) {
+        checkLimit(limit);
+        this.limit = Integer.valueOf(limit);
+    }
+
+    private void checkLimit(int limit) {
```

Comment:
Could you please use checkArgument from guava. Guava is repackaged in the jersey dependency

Reply:
Shouldn't here be BadRequestException instead of IllegalArgumentException?

Reply:
Thanks for the review!
@JoergAdler: now, offset and limit are checked by checkArgument  froom guava.
@Busoniu: if the SearchQueryParameterBean is used in the rest context, a ConstraintViolationException is thrown before the instance is created. The IllegalArgumentException should only be thrown, if a SearchQueryParameterBean is manually created .